### ISSUE TITLE
Behaviour of Recurring Transaction dialog not consistent with Single Transaction dialog - Amount Field Clearing

### ISSUE THREAD

**Issue by tactilis** [moneymanagerex/moneymanagerex#4787]
MMEX version:
 - [x] 1.5.16
 - [x] 1.5.17 Beta 1

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

A user entered value in the Amount field of the Recurring Transaction (RT) dialog is cleared if the transaction Type is changed to Transfer.

This is not consistent with the behaviour of the Single Transaction (ST) dialog, where the Amount field retains its value when the transaction Type is changed to Transfer.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Open the RT dialog
2. Enter a value in the Amount field
3. Realise that you should have set the Type to Transfer, so do it now
4. The Amount value is cleared (**bug**)

https://user-images.githubusercontent.com/4060092/177373361-39d60543-0726-4a2f-b243-fc563f930fbb.mp4

Expected result:
The behaviour should be as on the Single Transaction (ST) dialog, where the Amount field retains its value when the transaction Type is changed to Transfer.

**Comment by whalley** [comment]
Yes, there will be a few discrepancies between the dialogs until we unify them. I've fixed this one, but if we find any other minor discrepancies, I suggest we ignore them until we unify.

**Comment by tactilis** [comment]
@whalley  Understood.

I'll focus my observations (if any) on the current behaviour of the Single Transaction dialog and assume that Recurring will ride along when it is unified.
